app/actions/remote/groups.ts

```diff
@@ -1,104 +1,89 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {storeGroups} from '@actions/local/group';
 import {Client} from '@client/rest';
 import NetworkManager from '@managers/network_manager';
 
 import {forceLogoutIfNecessary} from './session';
 
-export const fetchGroupsForChannel = async (serverUrl: string, channelId: string) => {
+export const fetchGroupsForAutocomplete = async (serverUrl: string, query: string, fetchOnly = false) => {
     let client: Client;
     try {
         client = NetworkManager.getClient(serverUrl);
-        return client.getAllGroupsAssociatedToChannel(channelId);
-    } catch (error) {
-        forceLogoutIfNecessary(serverUrl, error as ClientErrorProps);
-        return {error};
-    }
-};
+        const response = await client.getGroups(query);
 
-export const fetchGroupsForMembership = async (serverUrl: string, userId: string) => {
-    let client: Client;
-    try {
-        client = NetworkManager.getClient(serverUrl);
-        return client.getAllGroupsAssociatedToMembership(userId);
-    } catch (error) {
-        forceLogoutIfNecessary(serverUrl, error as ClientErrorProps);
-        return {error};
-    }
-};
+        // Save locally
+        if (!fetchOnly) {
+            return await storeGroups(serverUrl, response);
+        }
```

Comment:
0/5 on preparing the records always (I would say it should return the api object, to avoid preparing objects that we may not store to the database), but if we keep this approach:
- The function should be called something more like "prepareGroups", since it does not always store.
- Instead of the if, you can pass the fetchOnly field to the prepareRecordsOnlyArgument.

Regarding being able to use both Group and GroupModel, you can see files like https://github.com/mattermost/mattermost-mobile/blob/gekidou/app/components/user_item/user_item.tsx where we use both User and UserModel.

Reply:
Ah good point - I already have `prepareGroups` so will use `storeGroups` just for store.